test/app/src/e2e/OutgoingCall.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { Button, Text, View } from 'react-native';
+import { Call, Voice } from '@twilio/voice-react-native-sdk';
+import { generateAccessToken } from '../tokenUtility';
+
+type TestStatus = 'not-started' | 'running' | 'success' | 'failed';
+
+type TestHook = () => { status: TestStatus; run: () => Promise<void> };
+
+async function settlePromise<T>(
+  p: Promise<T>
+): Promise<{ result: 'ok' | 'err'; value: T }> {
+  return p
+    .then((value) => ({ result: 'ok' as const, value }))
+    .catch((value) => ({ result: 'err' as const, value }));
+}
+
+const useOutgoingCallTest: TestHook = () => {
+  const testId = React.useMemo(() => Date.now(), []);
+  const voice = React.useMemo(() => new Voice(), []);
+  const accessToken = React.useMemo(() => generateAccessToken(), []);
+
+  const [status, setStatus] = React.useState<TestStatus>(() => 'not-started');
+
+  const run = React.useCallback(async () => {
+    setStatus('running');
+
+    const call = await voice.connect(accessToken);
+
+    const connectedPromise = new Promise<'connected'>((resolve) => {
+      call.on(Call.Event.Connected, () => {
+        console.log(testId, 'call event connected');
+        resolve('connected');
+      });
+    });
+
+    const connectFailurePromise = new Promise<'connectFailure'>((resolve) => {
+      call.on(Call.Event.ConnectFailure, (error) => {
+        console.log(testId, 'call event connectFailure', error);
+        resolve('connectFailure');
+      });
+    });
+
+    const callStatus = await Promise.any([
+      connectedPromise,
+      connectFailurePromise,
+    ]);
+
+    if (callStatus === 'connectFailure') {
+      setStatus('failed');
+      return;
+    }
+
+    await new Promise((resolve) => {
+      setTimeout(resolve, 5000);
```

Comment:
is the manual wait time still necessary? appium doesn't have a way to `.wait`?

Reply:
So appium does have a way to wait, but the way I set up the test such that each test is it's own UI and each hook is it's own test run time means that appium isn't in control here. At this point, it's app code, and the appium wait is more similar to how jest waits for tests to finish since appium is the test runner.